tests/unit/connect.c

```diff
@@ -92,6 +92,35 @@ int main(int argc, char *argv[])
   const char *user = getenv("MYSQL_USER");
   const char *pass = getenv("MYSQL_PASSWORD");
   const char *db = getenv("MYSQL_SCHEMA");
+  char error_expected[DRIZZLE_MAX_ERROR_SIZE];
+
+  // invalid host
+  test_connection_error("1.2.3.4", port, user, pass, db, DRIZZLE_RETURN_TIMEOUT,
+    "drizzle_wait:timeout reached", 3);
+
+  // invalid port
+  test_connection_error(host, 1234, user, pass, db,
+    DRIZZLE_RETURN_COULD_NOT_CONNECT, "drizzle_state_connect:could not connect",
+    -1);
+
+  // invalid user
+  sprintf(error_expected, "drizzle_check_unpack_error: Access denied for user "
```

Comment:
The pass is read from the env varr MYSQL_PASSWORD. The end of the error message is `(using password : YES )` or `(using password : NO )` depending on the value of that env var. 
However do see that using `sprintf` can be removed if we hardcode the value for `pass` in this case. 
This could be applied to the other cases as well and thus simplify the code. Seems like a better option since the connection parameters are somewhat irrelevant for these tests. 
Thus will hardcode the parameters.